src/lib/optimizer/abstract_syntax_tree/union_node.cpp

```diff
@@ -18,12 +18,19 @@ std::string UnionNode::description() const { return "[UnionNode] Mode: " + union
 
 std::string UnionNode::get_verbose_column_name(ColumnID column_id) const {
   Assert(left_child() && right_child(), "Need children to determine Column name");
+  Assert(column_id < left_child()->output_column_names().size(), "ColumnID out of range");
+  Assert(right_child()->output_column_names().size() == left_child()->output_column_names().size(),
+         "Input node mismatch");
```

Comment:
this seems like an odd place for this assertion. shouldn't this be checked way earlier, right after the children have been set?

Reply:
Asserting this elsewhere loosens the contract the Assert is making. There is no guarantee (apart from the model in our head, but that's precisely what's fallible and therefore needs asserting) that children won't change after `_on_child_changed()`.
So no, I think this Assertion is perfectly fine here... We could probably debate it if it would appear in another 10 places.